### PR TITLE
feat(console): add idp-initiated sso config page

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add idp-initiated SSO config page

image

image

## Testing

test locally

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/console/src/ds-components/Select/index.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ type Props<T> = {
   readonly options: Array<Option<T>>;
   readonly onChange?: (value?: T) => void;
   readonly isReadOnly?: boolean;
-  readonly error?: string | boolean;
+  readonly error?: string | boolean | ReactElement;
```
Comment (charIeszhao) [inline packages/console/src/ds-components/Select/index.tsx:29]: Use `ReactNode` instead? Since `ReaceNode` includes `string` and `boolean`

Comment (charIeszhao) [reply packages/console/src/ds-components/Select/index.tsx:29]: Or maybe if you want to be specific, like what you did later in Line 153, you can only allow `string | ReactElement` maybe?

File: packages/console/src/pages/EnterpriseSsoDetails/IdpInitiatedAuth/ConfigForm.tsx [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+      return t('enterprise_sso_details.idp_initiated_auth_config.empty_redirect_uris_error');
+    }
+  }, [defaultApplication, defaultApplicationRedirectUris.length, t]);
+
+  // Force set autoSendAuthorizationRequest to false if the default application is set to SPA
+  useEffect(() => {
+    if (defaultApplication?.type === ApplicationType.SPA) {
+      setValue('config.autoSendAuthorizationRequest', false);
+    }
+  }, [defaultApplication?.type, setValue]);
```
Comment (charIeszhao) [inline packages/console/src/pages/EnterpriseSsoDetails/IdpInitiatedAuth/ConfigForm.tsx:102]: This `useEffect` can be saved by writting the `setValue` in the `onChange` function, IMO

File: packages/console/src/pages/EnterpriseSsoDetails/IdpInitiatedAuth/ConfigForm.tsx [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+      const updated = await api
+        .put(buildIdpInitiatedAuthConfigEndpoint(ssoConnector.id), { json: payload })
+        .json<SsoConnectorIdpInitiatedAuthConfig>();
+      await mutateIdpInitiatedConfig(updated);
+      toast.success(t('general.saved'));
+      reset(parseResponseToFormData(updated));
+    }),
+    (errors) => {
+      console.log(errors);
```
Comment (charIeszhao) [inline packages/console/src/pages/EnterpriseSsoDetails/IdpInitiatedAuth/ConfigForm.tsx:134]: Toast?

File: packages/console/src/pages/EnterpriseSsoDetails/IdpInitiatedAuth/ConfigForm.tsx [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+
+      if (isSubmitting) {
+        return;
+      }
+
+      if (!isIdpInitiatedSsoEnabled || !config) {
+        await api.delete(buildIdpInitiatedAuthConfigEndpoint(ssoConnector.id));
+        await mutateIdpInitiatedConfig(undefined);
+        toast.success(t('general.saved'));
+        reset(parseResponseToFormData(undefined));
```
Comment (charIeszhao) [inline packages/console/src/pages/EnterpriseSsoDetails/IdpInitiatedAuth/ConfigForm.tsx:116]: Do you need to explicitly pass `undefined` to these functions?

File: packages/console/src/pages/EnterpriseSsoDetails/IdpInitiatedAuth/ConfigForm.tsx (hunk 3 above)
Comment (simeng-li) [reply packages/console/src/pages/EnterpriseSsoDetails/IdpInitiatedAuth/ConfigForm.tsx:134]: removed

File: packages/console/src/pages/EnterpriseSsoDetails/IdpInitiatedAuth/ConfigForm.tsx (hunk 4 above)
Comment (simeng-li) [reply packages/console/src/pages/EnterpriseSsoDetails/IdpInitiatedAuth/ConfigForm.tsx:116]: removed

**Review by charIeszhao — APPROVED** [review]

File: packages/console/src/pages/EnterpriseSsoDetails/IdpInitiatedAuth/ConfigForm.tsx (hunk 2 above)
Comment (simeng-li) [reply packages/console/src/pages/EnterpriseSsoDetails/IdpInitiatedAuth/ConfigForm.tsx:102]: `defaultApplication` is a computed value. Using `useEffect` to subscribe the changes is more straightforward.

**Pull request merged**